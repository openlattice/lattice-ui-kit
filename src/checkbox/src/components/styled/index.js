// @flow
import styled, { css } from 'styled-components';
import { NEUTRALS, PURPLES } from '../../../../colors';

const CheckboxLabel = styled.label`
  align-items: center;
  color: ${NEUTRALS[0]};
  display: inline-flex;
  font-size: 14px;
  font-weight: normal;
  margin: 10px 0;
  min-height: 20px;
  padding: 0 10px 0 30px;
  pointer-events: ${props => (props.readOnly ? 'none' : 'auto')};
  position: relative;
`;


type IndicatorProps = {
  checked :boolean;
};

const getAfterStyles = ({ checked } :IndicatorProps) => {
  if (!checked) {
    return css`
      display: none;
    `;
  }

  return css`
    content: '';
    position: absolute;
    left: 8px;
    top: 4px;
    width: 3px;
    height: 8px;
    border: solid white;
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
  `;
};

const CheckboxIndicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background: ${PURPLES[6]};
  border-radius: 3px;

  :after {
    ${getAfterStyles}
    }
  }
`;

const CheckboxInput = styled.input.attrs(() => ({
  type: 'checkbox'
}))`
  position: absolute;
  z-index: -1;
  opacity: 0;
  vertical-align: middle;
  
  :hover + ${CheckboxIndicator},
  :focus + ${CheckboxIndicator} {
    background: ${NEUTRALS[1]};
  }

  :checked + ${CheckboxIndicator} {
    background: ${PURPLES[2]};
  }

  :checked:hover + ${CheckboxIndicator},
  :checked:focus + ${CheckboxIndicator} {
    background: ${PURPLES[1]};
  }

  :disabled + ${CheckboxIndicator} {
    background: ${PURPLES[6]};
    cursor: not-allowed;
  }

  :checked:disabled + ${CheckboxIndicator} {
    background: ${NEUTRALS[2]};
  }
`;


export {
  CheckboxIndicator,
  CheckboxInput,
  CheckboxLabel,
};
