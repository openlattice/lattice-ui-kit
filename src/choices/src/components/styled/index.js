// @ flow
import CheckboxIndicator from './CheckboxIndicator';
import ChoiceLabel from './ChoiceLabel';
import RadioIndicator from './RadioIndicator';
import { CheckboxInput, RadioInput } from './ChoiceInputs';

export {
  CheckboxIndicator,
  CheckboxInput,
  ChoiceLabel,
  RadioIndicator,
  RadioInput
};
