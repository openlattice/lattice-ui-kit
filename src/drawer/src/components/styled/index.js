import styled, { css } from 'styled-components';

const getPosition = (props) => {
  const { side } = props;
  switch (side) {
    case 'right':
      return css`
        right: 0;
      `;

    case 'left':
    default:
      return css`
        left: 0;
      `;
  }
};

const DrawerWrapper = styled.div`
  position: fixed;
  background-color: white;
  height: 100%;
  width: 300px;
  top: 0;
  z-index: 900;
  box-shadow: 0 0 30px 0 rgba(0, 0, 0, 0.2);

  ${getPosition};
`;

export {
  // eslint-disable-next-line import/prefer-default-export
  DrawerWrapper
};
