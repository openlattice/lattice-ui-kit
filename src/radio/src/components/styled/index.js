// @flow
import styled, { css } from 'styled-components';
import { NEUTRALS, PURPLES, WHITE } from '../../../../colors';

const RadioLabel = styled.label`
  align-items: center;
  color: ${NEUTRALS[0]};
  display: inline-flex;
  margin: 10px 0;
  min-height: 20px;
  padding-left: 30px;
  pointer-events: ${props => (props.readOnly ? 'none' : 'auto')};
  position: relative;
`;

type IndicatorProps = {
  checked :boolean;
};

const getAfterStyles = ({ checked } :IndicatorProps) => {
  if (!checked) {
    return css`
      display: none;
    `;
  }

  return css`
    content: '';
    background-color: ${WHITE};
    border-radius: 50%;
    height: 8px;
    left: 6px;
    position: absolute;
    top: 6px;
    width: 8px;
  `;
};

const RadioIndicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background: ${PURPLES[6]};
  border-radius: 50%;

  :after {
    ${getAfterStyles}
  }
`;

const RadioInput = styled.input.attrs(() => ({
  type: 'radio'
}))`
  position: absolute;
  z-index: -1;
  opacity: 0;
  vertical-align: middle;

  :hover + div,
  :focus + div {
    background: ${NEUTRALS[1]};
  }

  :checked + div {
    background: ${PURPLES[2]};
  }

  :checked:hover + div,
  :checked:focus + div {
    background: ${PURPLES[1]};
  }

  :disabled + div {
    background: ${PURPLES[6]};
    cursor: not-allowed;
  }

  :checked:disabled + div {
    background: ${NEUTRALS[2]};
  }
`;

export {
  RadioLabel,
  RadioInput,
  RadioIndicator,
};
