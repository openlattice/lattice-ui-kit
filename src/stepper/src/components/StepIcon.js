// @flow
import React from 'react';

import { faCheck } from '@fortawesome/pro-regular-svg-icons';
import { faCircle } from '@fortawesome/pro-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { IconLayer, StepIndex } from './styled';

import {
  NEUTRALS,
  PURPLE,
  WHITE
} from '../../../colors';

type Props = {
  active :boolean;
  className ? :string;
  complete :boolean;
  index :number;
};

const StepIcon = ({
  active,
  className,
  complete,
  index
} :Props) => {
  const styledClassName = className ? `${className} fa-layers` : 'fa-layers';
  return (
    <IconLayer className={styledClassName}>
      <FontAwesomeIcon icon={faCircle} color={(active || complete) ? PURPLE.P300 : NEUTRALS[1]} />
      {
        complete
          ? <FontAwesomeIcon icon={faCheck} color={WHITE} transform="shrink-7" />
          : <StepIndex className="fa-layers-text fa-inverse">{index}</StepIndex>
      }
    </IconLayer>
  );
};

StepIcon.defaultProps = {
  className: undefined
};

export default StepIcon;
