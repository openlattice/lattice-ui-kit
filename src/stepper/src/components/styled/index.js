import styled, { css } from 'styled-components';
import { NEUTRALS } from '../../../../colors';

const IconLayer = styled.span`
  font-size: 24px;
  margin-right: 8px;
`;

const getDividerOrientationStyles = ({ vertical }) => {
  if (vertical) {
    return css`
      min-height: 20px;
      border-left: 2px solid ${NEUTRALS[4]};
       /* step padding + half icon - 1 */
      margin-left: 19px;
    `;
  }

  return css`
    border-top: 2px solid ${NEUTRALS[4]};
  `;
};

const StepDivider = styled.div`
  flex: 1 1 auto;
  ${getDividerOrientationStyles};
`;

const StepIndex = styled.span`
  font-size: 14px;
  font-weight: bold;
`;

const StepWrapper = styled.div`
  display: flex;
  flex: 0 1 auto;
  align-items: center;
  padding: 10px;
  cursor: ${props => props.onClick && 'pointer'};
`;

const getStepperOrientationStyles = ({ vertical }) => {
  if (vertical) {
    return css`
      align-items: flex-start;
      flex-direction: column;
      justify-content: start;
    `;
  }

  return css`
    flex-direction: row;
    justify-content: space-around;
  `;
};

const StepperWrapper = styled.div`
  align-items: center;
  display: flex;
  flex: 1 0 auto;
  ${getStepperOrientationStyles};
`;

export {
  IconLayer,
  StepDivider,
  StepIndex,
  StepperWrapper,
  StepWrapper,
};
