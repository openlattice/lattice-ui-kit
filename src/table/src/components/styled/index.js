// @flow
import styled from 'styled-components';

import { NEUTRALS, WHITE } from '../../../../colors';
import { getStickyPosition } from '../../../../utils/StyleUtils';

const StyledTable = styled.table`
  background-color: ${WHITE};
  border-collapse: collapse;
  border: none;
  table-layout: fixed;
  width: 100%;
`;

const PaginationWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: flex-end;
  width: 100%;
`;

const Cell = styled.td((props :Object) => ({
  backgroundColor: props.as === 'th' ? NEUTRALS[6] : 'inherit',
  cursor: props.onClick ? 'pointer' : 'auto',
  padding: '10px 10px',
  textAlign: 'left',
  wordWrap: 'break-word',
  ...props.cellStyle
}));

Cell.displayName = 'Cell';

const StyledRow = styled.tr`
  background-color: ${WHITE};
  border-bottom: 1px solid ${NEUTRALS[4]};

  /* position: sticky doesn't work with thead/tr. Move to all cells within */
  td, th {
    ${getStickyPosition}
  };
`;

const RowPerPageWrapper = styled.div`
  font-size: 14px;
  margin-right: 10px;
  width: 75px;
`;

export {
  Cell,
  PaginationWrapper,
  RowPerPageWrapper,
  StyledRow,
  StyledTable,
};
