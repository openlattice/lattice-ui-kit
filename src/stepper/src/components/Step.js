// @flow

import React from 'react';
import type { ChildrenArray } from 'react';

import StepIcon from './StepIcon';
import { StepWrapper } from './styled';

type Props = {
  index :number;
  active :boolean;
  complete :boolean;
  children :ChildrenArray<any>;
  onClick ? :(e :SyntheticEvent<HTMLDivElement>) => void;
};

const Step = ({
  active,
  complete,
  children,
  index,
  onClick
} :Props) => (
  <StepWrapper onClick={onClick}>
    <StepIcon
        index={index}
        active={active}
        complete={complete} />
    {children}
  </StepWrapper>
);

Step.defaultProps = {
  onClick: undefined
};

export default Step;
