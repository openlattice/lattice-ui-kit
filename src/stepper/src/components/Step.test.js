import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Step from './Step';
import StepIcon from './StepIcon';
import { StepWrapper, StepLabel } from './styled';

describe('Step', () => {
  test('should match snapshot', () => {
    const wrapper = mount(<Step index={0}>Step 1</Step>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('styles', () => {
    test('default styles', () => {
      const wrapper = mount(<Step index={0}>Step 1</Step>);
      expect(wrapper.find(StepWrapper)).toHaveStyleRule('display', 'flex');
      expect(wrapper.find(StepWrapper)).toHaveStyleRule('align-items', 'center');
      expect(wrapper.find(StepWrapper)).toHaveStyleRule('padding', '10px');
    });

    test('onClick adds pointer styles', () => {
      const wrapper = mount(<Step index={0} onClick={() => {}}>Step 1</Step>);
      expect(wrapper.find(StepWrapper)).toHaveStyleRule('cursor', 'pointer');
    });

    test('StepLabel should apply font-weight if active', () => {
      const wrapper = mount(<Step index={0} active>Step 1</Step>);
      expect(wrapper.find(StepLabel)).toHaveStyleRule('font-weight', '600');
    });
  });

  test('render StepIcon and StepLabel', () => {
    const wrapper = shallow(<Step index={0}>Step 1</Step>);
    expect(wrapper.find(StepIcon).prop('index')).toBe(0);
    expect(wrapper.find(StepLabel).text()).toBe('Step 1');
  });

});
