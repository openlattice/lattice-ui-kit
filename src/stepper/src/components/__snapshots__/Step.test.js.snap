// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step should match snapshot 1`] = `
.c1 {
  font-size: 24px;
  min-width: 24px;
}

.c2 {
  font-size: 14px;
  font-weight: bold;
}

.c3 {
  margin-left: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
}

<Step
  index={0}
>
  <styled__StepWrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__StepWrapper-g8vnjr-4",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "display:flex;align-items:center;padding:10px;cursor:",
              [Function],
              ";",
            ],
          },
          "displayName": "styled__StepWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__StepWrapper-g8vnjr-4",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <StepIcon
          index={0}
        >
          <styled__IconLayer
            className="fa-layers"
          >
            <StyledComponent
              className="fa-layers"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styled__IconLayer-g8vnjr-0",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "font-size:24px;min-width:24px;",
                    ],
                  },
                  "displayName": "styled__IconLayer",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styled__IconLayer-g8vnjr-0",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <span
                className="c1 fa-layers"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  color="#8b8b90"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f111",
                        "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z",
                      ],
                      "iconName": "circle",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-circle fa-w-16 "
                    color="#8b8b90"
                    data-icon="circle"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
                <styled__StepIndex
                  className="fa-layers-text fa-inverse"
                >
                  <StyledComponent
                    className="fa-layers-text fa-inverse"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "styled__StepIndex-g8vnjr-2",
                          "isStatic": false,
                          "lastClassName": "c2",
                          "rules": Array [
                            "font-size:14px;font-weight:bold;",
                          ],
                        },
                        "displayName": "styled__StepIndex",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "styled__StepIndex-g8vnjr-2",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="c2 fa-layers-text fa-inverse"
                    >
                      0
                    </span>
                  </StyledComponent>
                </styled__StepIndex>
              </span>
            </StyledComponent>
          </styled__IconLayer>
        </StepIcon>
        <styled__StepLabel>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styled__StepLabel-g8vnjr-3",
                  "isStatic": false,
                  "lastClassName": "c3",
                  "rules": Array [
                    "margin-left:8px;font-weight:",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "styled__StepLabel",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styled__StepLabel-g8vnjr-3",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="c3"
            >
              Step 1
            </span>
          </StyledComponent>
        </styled__StepLabel>
      </div>
    </StyledComponent>
  </styled__StepWrapper>
</Step>
`;
